covariance/covarianceXsec.h

```diff
@@ -12,42 +12,52 @@
 // MaCh3 includes
 #include "covarianceBase.h"
 #include "samplePDF/Structs.h"
-#include "throwParms/ThrowParms.h"
+
+#include "yaml-cpp/yaml.h"
 
 class covarianceXsec : public covarianceBase {
 
   public:
-  covarianceXsec(const char *name, const char *file, double threshold=-1,int firstpcapar=-999,int lastpcapar=-999);
+  covarianceXsec(const char *name, const char *file, double threshold=-1,int FirstPCAdpar=-999,int LastPCAdpar=-999);
+  covarianceXsec(const char *YAMLFile);
     ~covarianceXsec();
 
     // Print information about the whole object once it is set
     void Print();
 
-    void throwNominal(bool nomValues=true, int seed = 0);
-
     // General Getter functions not split by detector
-    double GetLikelihood();
-    double GetParamUpperBound(int i) {return xsec_param_ub_a[i];}
-    double GetParamLowerBound(int i) {return xsec_param_lb_a[i];}
-    double GetParamPrior(int i)      {return xsec_param_prior_a[i];}
-    int  GetXSecParamID(int i, int j) const {return xsec_param_id_a[i][j];}
-    const std::string & GetParameterName(int i) const {return xsec_param_names[i];}
-    int    GetNumParams()               {return nPars;}
-    const char* GetParName(int i) const {return xsec_param_names[i].c_str();}
-
-    bool isParFlux(int i){
+	// ETA - a lot of these can go... they're just duplications from the base
+	// class.
+    const double GetParamUpperBound(const int i) {return xsec_param_ub_a[i];}
+    const double GetParamLowerBound(const int i) {return xsec_param_lb_a[i];}
+    const double GetParamPrior(const int i)      {return xsec_param_prior_a[i];}
+    const int  GetXSecParamID(const int i, const int j) const {return xsec_param_id_a[i][j];}
```

Comment:
Agreed and we can put inline ;)

Reply:
Just a small comment on all these `const double/int ...` functions. I don't know what your intent is there with the `const` qualifier, but it currently does not mean anything to the compiler. If the intent is to say that the function does not affect the class object, `const` should be moved at the end of the signature. Otherwise, as you are returning by value I'm pretty sure it has no impact/meaning.

Reply:
I've removed a lot of the consts from the getters just by siting through the unqualified warnings. I haven't made them inline but I dont' think this is so important